daemon/list.go

```diff
@@ -499,14 +499,14 @@ func includeContainerInList(container *container.Container, ctx *listContext) it
 	networkExist := fmt.Errorf("container part of network")
 	if ctx.filters.Include("network") {
 		err := ctx.filters.WalkValues("network", func(value string) error {
-			if _, ok := container.NetworkSettings.Networks[value]; ok {
-				return networkExist
-			}
 			for _, nw := range container.NetworkSettings.Networks {
 				if nw.EndpointSettings == nil {
 					continue
 				}
-				if nw.NetworkID == value {
+				if strings.HasPrefix(nw.NetworkID, value) {
+					return networkExist
+				}
+				if strings.HasPrefix(container.HostConfig.NetworkMode.NetworkName(), value) {
```

Comment:
@thaJeztah  
If we removed match on networkName, it wont work if net1 and net111 network preset and filtered with netonly as commented as above.
Should we remove this also ?